packages/core/src/components/HierarchyPicker/index.tsx

```diff
@@ -0,0 +1,262 @@
+
+        if (!item) {
+          return '';
+        }
+
+        return i < chosen.length - 1 ? item.items || [] : item.label || item.name;
+      }
+
+      return '';
+    }, this.props.items);
+
+    return typeof label === 'string' ? label : '';
+  };
+
+  boundFormatter = (chosen: TreePath) => this.props.formatter!(chosen, this.getLabel);
+
+  toggle = () => {
+    if (this.props.disabled) {
+      return;
+    }
+
+    this.setState(
+      state => ({ open: !state.open }),
+      () => {
+        const { open } = this.state;
+        if (open) {
+          this.props.onPickerToggle!(true);
+        } else {
+          this.handleClose();
+        }
+      },
+    );
+  };
+
+  private handleClose = () => {
+    this.props.onPickerToggle!(false);
+    this.setState({ open: false });
+
+    const el = this.ref.current;
+
+    if (el) {
+      el.focus();
+    }
+  };
+
+  private handleClick = () => {
+    this.toggle();
+  };
+
+  private handleKeyDown = (event: React.KeyboardEvent<HTMLDivElement>) => {
+    if (event.target !== this.ref.current) {
+      return;
+    }
+
+    switch (event.key) {
+      case ENTER:
+      case SPACE:
+        event.preventDefault();
+        this.toggle();
+        break;
+      default:
+        break;
+    }
+  };
+
+  render() {
+    const {
+      styles,
+      disabled,
+      invalid,
+      children,
+      searchPlaceholder,
+      noResultsLabel,
+      ...passThruProps
+    } = this.props;
+    const { chosen } = passThruProps;
+    const { open } = this.state;
+
+    return (
+      <div>
+        <div
+          {...css(
+            styles.selectlike,
+            disabled ? styles.selectlike_disabled : styles.selectlike_enabled,
+            invalid && styles.selectlike_invalid,
+          )}
+          tabIndex={disabled ? -1 : 0}
+          role="button"
+          onKeyDown={this.handleKeyDown}
+          onClick={this.handleClick}
+          ref={this.ref}
+        >
+          {children || <Text>{this.boundFormatter(chosen || [])}</Text>}
+
+          <span {...css(styles.arrow)}>
+            <IconCaretDown decorative size="1.5em" />
+          </span>
+        </div>
+
+        <Overlay open={open} onClose={this.handleClose}>
+          <Picker
+            {...passThruProps}
+            formatter={this.boundFormatter}
+            onClose={this.handleClose}
+            searchPlaceholder={
+              searchPlaceholder ||
+              T.phrase('Search', {}, 'Search for a topic within the hierarchy picker')
+            }
+            noResultsLabel={
+              noResultsLabel ||
+              T.phrase(
+                'No results',
+                {},
+                'Label to display when no results are found for hierarchy picker search',
+              )
+            }
+          />
+        </Overlay>
+      </div>
+    );
+  }
+}
+
+export default withStyles(({ color, unit, ui, pattern }) => ({
+  selectlike: {
```

Comment:
We should probably inherit styles from `buildInputStyles`, instead of doing this custom.